scylla/src/transport/load_balancing/default.rs

```diff
@@ -68,6 +69,21 @@ or refrain from preferring datacenters (which may ban all other datacenters, if
             }
         }
         if let Some(ts) = &routing_info.token_with_strategy {
+            // Try to pick some alive local rack random replica.
+            // If preferred rack is not specified, try to pick local DC replica.
+            if let Some(_local_rack) = &self.preferred_rack {
```

Comment:
`if let` is a bit superfluous here, you are creating an unused `_local_rack` variable. `if self.preferred_rack.is_some()` would be a better choice there probably.